bpf/Makefile.bpf

```diff
@@ -1,7 +1,7 @@
 # Copyright 2017-2020 Authors of Cilium
 # SPDX-License-Identifier: Apache-2.0
 
-FLAGS := -I$(ROOT_DIR)/bpf/include -I$(ROOT_DIR)/bpf -D__NR_CPUS__=$(shell nproc --all) -O2 -g
+FLAGS := -I$(ROOT_DIR)/bpf/include -I$(ROOT_DIR)/bpf -D__NR_CPUS__=$(shell cut -d- -f2 /sys/devices/system/cpu/possible) -O2 -g
```

Comment:
I don't think this will work reliably either. On my machine, this yields `7` (from `0-7`), even though I have 8 cores including SMT. Secondly, I also believe that this does not handle cases where `possible` returns multiple ranges

I believe some machines (e.g. with NUMA) it is possible that `possible` will return multiple ranges, e.g. (`0-4,8-12`). This snippet does not handle that case.

In addition, I think single core systems will also just report a single number.

I think we probably have to ship either a Go binary or a small script to parse `/sys/devices/system/cpu/possible` the same way we do it in 
https://github.com/cilium/cilium/blob/2ff7193bb2301523af39695646ac4eb2a1294f17/pkg/common/utils.go#L170

Reply:
I'll rework that on Monday.

Reply:
I've updated the branch to simply pass the value we use on Golang side to `bpf/init.sh`. The two other places that reference `__NR_CPUS__` are only used for unit tests, Sparse, and compile testing, so I don't think it matters that we have the same value as Golang side there. We actually always make sure to use different map names in those places to avoid conflicts with the actual maps used at runtime.